cumulus_library/studies/discovery/code_definitions.py

```diff
@@ -0,0 +1,53 @@
+# A collection of codes & codeableConcepts to extract available codes from.
+# is_array and is_bare_code are assumed to be false unless explicitly set true.
```

Comment:
nit: Maybe a quick comment that they both also can't be true - which is presumably obvious, but still.

And/or maybe like a light explanation of each option:
- `is_array`: array of [CodeableConcepts](https://www.hl7.org/fhir/R4/datatypes.html#CodeableConcept)
- `is_bare_code`: direct [Coding](https://www.hl7.org/fhir/R4/datatypes.html#Coding), not a full CodeableConcept
- neither: one CodeableConcept

As/if/when you add future field types, would you add more booleans? Maybe a string/enum field makes sense here instead of a boolean tree. Especially since the default option isn't an intuitive lack of the other booleans being true, but like, just a third option.